src/arch/xtensa/include/xtensa/config/core.h

```diff
@@ -36,6 +36,19 @@
 #ifndef XTENSA_CONFIG_CORE_H
 #define XTENSA_CONFIG_CORE_H
 
+/* Zephyr/XCC specific definitions */
+#if defined(__ZEPHYR__)
+
+#ifndef __UINT32_C
+#define __UINT32_C(x) x ## U
+#endif
+
+#ifndef UINT32_C
+#define UINT32_C(x) __UINT32_C(x)
+#endif
```

Comment:
This `#ifndef #define` pattern is at best a necessary evil because the definition can change depending the `#include` order (recent example: 2512698098ed) and the direction the wind blows.

So can you transfer some of the information and justification from the commit message to the comment here?

Reply:
The reason I had to put it in here in the first place, especially with the __ZEPHYR__ guard, is because of the include order. On XTOS builds (including i.MX8) without that guard this is included before newlib and newlib does _not_ do this, which leads to conflicts.

I will copy the comment and resubmit, maybe today but if not on Tuesday.

Reply:
> On XTOS builds (including i.MX8) without that guard this is included before newlib and newlib does not do this, which leads to conflicts.

What sort of conflicts? This is coming up again in non-Zephyr build https://github.com/thesofproject/sof/pull/9351#discussion_r1707489595 ...

Maybe @keith-packard or @stephanosio could help us get out of this libc / toolchain nightmare?